backend/x11/output.c

```diff
@@ -475,6 +475,19 @@ static bool output_set_cursor(struct wlr_output *wlr_output,
 	if (texture != NULL) {
 		width = texture->width * wlr_output->scale / scale;
 		height = texture->height * wlr_output->scale / scale;
+
+		if (hotspot_x < 0) {
```

Comment:
Shouldn't we do this after transforming the hotspot?

Reply:
You tell me :)

The logic I was going under was that hotspot_x/y are in the texture coordinate space, so it's fine to clamp them to texture->width/height. It wasn't completely obvious to me what the limits of the post-transform hotspot would be -- I guess width/height?

Reply:
Looking at the code again, I think you're right and the hotspot is in texture coords.